app/src/main/java/com/example/nextgen/splashScreen/SplashScreenActivity.kt

```diff
@@ -0,0 +1,28 @@
+package com.example.nextgen.splashScreen
+
+import android.content.Intent
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.lifecycleScope
+import com.example.nextgen.R
+import com.example.nextgen.signup.SignupSigninActivity
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+
+class SplashScreenActivity : AppCompatActivity() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_splash_screen)
+
+        lifecycleScope.launch {
+            delay(3000) // 3-second delay
+            navigateToMainActivity()
+        }
+    }
+
+    private fun navigateToMainActivity() {
+        startActivity(Intent(this, SignupSigninActivity::class.java))
```

Comment:
Redirect user to main screen here itself based on sign in status.

Reply:
Remove redirection logic to main screen  in `SignUpSignInActivity` and place it in Splash activity. Handle both main and signin screen redirection logic in Splash activity.